internal/publisher/core_test.go

```diff
@@ -0,0 +1,82 @@
+package publisher
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	brokerstoremock "github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	mockMB "github.com/razorpay/metro/pkg/messagebroker/mocks"
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+	"github.com/rs/xid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCore_Publish(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	mockBrokerStore := brokerstoremock.NewMockIBrokerStore(ctrl)
+	producer := mockMB.NewMockProducer(ctrl)
+	publisherCore := NewCore(mockBrokerStore)
+	req := getDummyPublishRequest()
+	msgID := xid.New().String()
+
+	tests := []struct {
+		name     string
+		req      *metrov1.PublishRequest
+		msgID    string
+		expected []string
+		wantErr  bool
+		err      error
+	}{
+		{
+			name:     "Message publishing with no errors",
+			req:      req,
+			msgID:    msgID,
```

Comment:
```suggestion
			msgID:    xid.New().String(),
```

Reply:
can't do, need to use same value of msgID in msgID and expected field